provider/pkg/await/await.go

```diff
@@ -208,13 +210,18 @@ func Creation(c CreateConfig) (*unstructured.Unstructured, error) {
 		if metadata.SkipAwaitLogic(c.Inputs) {
 			logger.V(1).Infof("Skipping await logic for %v", c.Inputs.GetName())
 		} else {
+			var namespace = c.Inputs.GetNamespace()
+			if namespace == "" {
+				namespace = metav1.NamespaceDefault
+			}
 			if awaiter.awaitCreation != nil {
 				conf := createAwaitConfig{
 					host:              c.Host,
 					ctx:               c.Context,
 					urn:               c.URN,
 					initialAPIVersion: c.InitialAPIVersion,
 					clientSet:         c.ClientSet,
+					informerFactory:   dynamicinformer.NewFilteredDynamicSharedInformerFactory(c.ClientSet.GenericClient, 60*time.Second, namespace, nil),
```

Comment:
I just noticed that the resync is from the local cache and doesn't check the authoritative source. Since all of the awaiters are running locally to the cache anyway, we should be able to disable the resync.

> The resync operation consists of delivering to the handler an update notification for every object in the informer's local cache; it does not add any interactions with the authoritative storage.

Reply:
I have removed the informer factory from the await config. I initially dropped the resync but I think it might be best to keep it there to allow us to ensure the await logic gets kicked periodically anyway which seems desirable. In a way this gives us polling for the await logic without actually hitting the API server so the cost of having this is very low.